app/views/two_factor_authentication/recovery_code_verification/show.html.slim

```diff
@@ -4,11 +4,15 @@ h1.h3.my0 = t('devise.two_factor_authentication.recovery_code_header_text')
 p.mt-tiny.mb0 = t('devise.two_factor_authentication.recovery_code_prompt')
 
 = form_tag(:login_two_factor_recovery_code, method: :post, role: 'form', class: 'mt3 sm-mt4') do
-  .mb2
-    = label_tag 'code',
-      t('simple_form.required.html') + t('forms.two_factor.recovery_code'),
-      class: 'block bold'
-    = block_text_field_tag :code, '', required: true, autocomplete: 'off'
+  label.block.bold#personal-key-label
+    = t('simple_form.required.html') + t('forms.two_factor.recovery_code')
+
+  - ENV['recovery_code_length'].to_i.times do |_|
```

Comment:
1. Let's use `Figaro.env` to access this value instead of ENV directly? It's easier to stub in tests and more consistent with our usage
2. If we're not using the argument to the block we can drop it (ex `do` with no `|_|` after it)